circuit-retained/src/commonMain/kotlin/com/slack/circuit/retained/RetainedStateRegistry.kt

```diff
@@ -144,8 +137,30 @@ internal class RetainedStateRegistryImpl(retained: MutableMap<String, List<Any?>
   }
 
   override fun forgetUnclaimedValues() {
-    // Notify any RememberObservers that it has been forgotten
-    retained.asSequence().filterIsInstance<RememberObserver>().forEach { it.onForgotten() }
+    clearRetained()
+  }
+
+  /** Should this be public API on RetainedStateRegistry? */
+  internal fun clear() {
+    // First we saveAll, which flattens down the value providers to our retained list
+    saveAll()
+    // Now we clear the retained list
+    clearRetained()
+  }
+
+  private fun clearRetained() {
+    fun clearValue(value: Any?) {
+      when (value) {
+        // If we get a RetainedHolder value, need to unwrap and call again
+        is RetainedValueHolder<*> -> clearValue(value.value)
+        // Dispatch the call to nested registries
+        is RetainedStateRegistryImpl -> value.clearRetained()
```

Comment:
Does using the impl limit folks' ability to bring their own registry impl? Should we move `clearRetained()` into the interface?

Reply:
Maybe! I'd probably move `clear()` instead (see the comment on that function).

Reply:
Cool, do you want to try it in this PR? If not I can try it separately

Reply:
Tidied up. We already have this function, it's just called `forgetUnclaimedValues`.